tensor_parallel.py

```diff
@@ -0,0 +1,315 @@
+import torch
+import torch.distributed as dist
+from torch.testing._internal.common_utils import TestCase
+from transformers import logging
+from typing import Any, Dict, List, Optional, Tuple, Union
+from transformers.cache_utils import Cache
+
+logger = logging.get_logger(__name__)
+
+device = "cuda" if torch.cuda.is_available() else "cpu" 
+tp_size = 1
```

Comment:
This is a bad way to write your code, `tp_size` should come from the user arguments and why should it kept as a global variable?